x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elser/ElserMlNodeService.java

```diff
@@ -0,0 +1,167 @@
+        boolean throwOnUnknownFields,
+        String modelId,
+        TaskType taskType,
+        Map<String, Object> settings
+    ) {
+        Map<String, Object> serviceSettingsMap = removeFromMapOrThrowIfNull(settings, Model.SERVICE_SETTINGS);
+        Map<String, Object> taskSettingsMap = removeFromMapOrThrowIfNull(settings, Model.TASK_SETTINGS);
+
+        var serviceSettings = serviceSettingsFromMap(serviceSettingsMap);
+        var taskSettings = taskSettingsFromMap(taskType, taskSettingsMap);
+
+        if (throwOnUnknownFields == false) {
+            throwIfNotEmptyMap(settings);
+            throwIfNotEmptyMap(serviceSettingsMap);
+            throwIfNotEmptyMap(taskSettingsMap);
+        }
+
+        return new ElserMlNodeModel(modelId, taskType, NAME, serviceSettings, taskSettings);
+    }
+
+    private final OriginSettingClient client;
+
+    public ElserMlNodeService(Client client) {
+        this.client = new OriginSettingClient(client, ClientHelper.INFERENCE_ORIGIN);
+    }
+
+    @Override
+    public Model parseConfigStrict(String modelId, TaskType taskType, Map<String, Object> config) {
+        return parseConfig(true, modelId, taskType, config);
+    }
+
+    @Override
+    public ElserMlNodeModel parseConfigLenient(String modelId, TaskType taskType, Map<String, Object> config) {
+        return parseConfig(false, modelId, taskType, config);
+    }
+
+    @Override
+    public void start(Model model, ActionListener<Boolean> listener) {
+        if (model instanceof ElserMlNodeModel == false) {
+            listener.onFailure(new IllegalStateException("not an elser model")); // TODO
+            return;
+        }
+
+        if (model.getTaskType() != TaskType.SPARSE_EMBEDDING) {
+            listener.onFailure(new IllegalStateException("not an elser model")); // TODO
+            return;
+        }
+
+        var elserModel = (ElserMlNodeModel) model;
+        var serviceSettings = elserModel.getServiceSettings();
+
+        var startRequest = new StartTrainedModelDeploymentAction.Request(ELSER_V1_MODEL, model.getModelId());
+        startRequest.setNumberOfAllocations(serviceSettings.getNumAllocations());
+        startRequest.setThreadsPerAllocation(serviceSettings.getNumThreads());
+        startRequest.setWaitForState(STARTED);
+
+        client.execute(
+            StartTrainedModelDeploymentAction.INSTANCE,
+            startRequest,
+            ActionListener.wrap(r -> listener.onResponse(Boolean.TRUE), listener::onFailure)
+        );
+    }
+
+    @Override
+    public void infer(
+        String modelId,
+        TaskType taskType,
+        String input,
+        Map<String, Object> config,
+        ActionListener<InferenceResult> listener
+    ) {
+
+        if (taskType != TaskType.SPARSE_EMBEDDING) {
+            listener.onFailure(
+                new ElasticsearchStatusException(
+                    "The Elser ML Node service does not support task type [{}]",
```

Comment:
In years to come it's going to sound very strange talking about an "ML Node service", when serverless doesn't have nodes at all (officially at least). It's probably worth one more discussion on naming before committing to this name. If the team can think of something more future proof then hopefully IntelliJ's renaming features mean it won't be too hard to rename at this stage.

Reply:
I've taken "ML Node" out of the error message, the exact service name is still TBD.